src/Plugin/Field/FieldWidget/ColorSchemeWidget.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\oe_color_scheme\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Extension\ThemeExtensionList;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\WidgetBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Theme\ThemeManagerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Plugin implementation of the 'Color Scheme' widget.
+ *
+ * @FieldWidget(
+ *   id = "oe_color_scheme_widget",
+ *   label = @Translation("Color Scheme widget"),
+ *   field_types = {
+ *     "oe_color_scheme"
+ *   }
+ * )
+ */
+class ColorSchemeWidget extends WidgetBase {
+
+  /**
+   * The theme manager.
+   *
+   * @var \Drupal\Core\Theme\ThemeManagerInterface
+   */
+  protected $themeManager;
```

Comment:
The new coding standards say that since D9 we should [use typehinting](https://www.drupal.org/docs/develop/standards/php/php-coding-standards#s-parameter-and-return-typehinting) wherever possible, and [omit @var](https://www.drupal.org/docs/develop/standards/php/api-documentation-and-comment-standards#var)